### PR TITLE
Convert traffic_cache_tool to use ts_file

### PR DESCRIPTION
**Pull request by chitianhao**
`File.cc` and `File.h` inside traffic_cache_tool is removed and the new library `ts_file` is used.
This also fixes the segfault I found a while ago when used on Mac.

### CODE REVIEW

File: src/traffic_cache_tool/CacheDefs.cc [hunk 1]
```diff
@@ -23,6 +23,8 @@
 
 #include "CacheDefs.h"
 #include <iostream>
+#include <fcntl.h>
```
Comment (SolidWallOfCode) [inline src/traffic_cache_tool/CacheDefs.cc]: If this is needed, something is wrong with `ts_file.h`.

File: src/traffic_cache_tool/CacheDefs.cc [hunk 2]
```diff
@@ -216,7 +218,7 @@ Stripe::Chunk::clear()
 
 Stripe::Stripe(Span *span, Bytes start, CacheStoreBlocks len) : _span(span), _start(start), _len(len)
 {
-  ts::bwprint(hashText, "{} {}:{}", span->_path.path(), _start.count(), _len.count());
+  ts::bwprint(hashText, "{} {}:{}", span->_path.string(), _start.count(), _len.count());
```
Comment (SolidWallOfCode) [inline src/traffic_cache_tool/CacheDefs.cc]: This is wrong - we need to update either `ts::file` or `BufferWriterFormat` to make this unneeded. I have another PR up that adds the `view` method to `ts::file::path` which should be used to support this.

File: src/traffic_cache_tool/CacheDefs.h [hunk 3]
```diff
@@ -276,11 +275,11 @@ class DFA;
...
 
 struct url_matcher {
-  url_matcher(ts::FilePath const &path) // file contains a list of regex
+  url_matcher(ts::file::path const &path) // file contains a list of regex
   {
-    ts::BulkFile cfile(path);
-    if (cfile.load() == 0) {
-      ts::TextView fileContent = cfile.content();
+    std::error_code ec;
+    ts::TextView fileContent = ts::file::load(path, ec);
```
Comment (SolidWallOfCode) [inline src/traffic_cache_tool/CacheDefs.h]: This won't work - where is the storage for the file contents?

File: src/traffic_cache_tool/CacheTool.cc [hunk 4]
```diff
@@ -538,26 +538,25 @@ Cache::loadSpanConfig(FilePath const &path)
...
   static const ts::TextView TAG_VOL("url");
   ts::URLparser parser;
   Errata zret;
 
-  ts::BulkFile cfile(path);
-  if (0 == cfile.load()) {
-    ts::TextView content = cfile.content();
-
+  std::error_code ec;
+  ts::TextView content = ts::file::load(path, ec);
```
Comment (SolidWallOfCode) [inline src/traffic_cache_tool/CacheTool.cc]: This won't work either.

File: src/traffic_cache_tool/CacheTool.cc [hunk 5]
```diff
@@ -721,7 +723,7 @@ Span::loadDevice()
 #endif
     ;
 
-  ats_scoped_fd fd(_path.open(flags));
+  ats_scoped_fd fd(!_path.empty() ? ::open(_path.c_str(), flags) : ats_scoped_fd());
 
   if (fd.isValid()) {
     if (ink_file_get_geometry(fd, _geometry)) {
```
Comment (SolidWallOfCode) [inline src/traffic_cache_tool/CacheTool.cc]: Outside scope - should look at moving this file geometry to `ts::file`.

File: src/traffic_cache_tool/CacheTool.cc [hunk 6]
```diff
@@ -1255,7 +1257,7 @@ Check_Freelist(std::string devicePath)
   if ((err = cache.loadSpan(SpanFile))) {
     cache.dumpSpans(Cache::SpanDumpDepth::SPAN);
     for (auto sp : cache._spans) {
-      if (devicePath.size() > 0 && 0 == strncmp(sp->_path.path(), devicePath.data(), devicePath.size())) {
+      if (devicePath.size() > 0 && 0 == strncmp(sp->_path.c_str(), devicePath.data(), devicePath.size())) {
```
Comment (SolidWallOfCode) [inline src/traffic_cache_tool/CacheTool.cc]: This is where the `ts::path::view` method would win - then it would be `&& sp->_path.view() == devicePath`.

**Comment by SolidWallOfCode** [comment]
#4406 has been merged, time to update this.

**Comment by chitianhao** [comment]
Updated. @SolidWallOfCode

**Comment by SolidWallOfCode** [comment]
[approve ci rat]

**Comment by SolidWallOfCode** [comment]
[approve ci]

**Comment by SolidWallOfCode** [comment]
[approve ci freebsd]

**Pull request merged**